### PR TITLE
Recoil overhaul [RTM]

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Recoil was so far very much unfinished, so I have decided to use some spare time and have some fun with it.
Copious alcohol, neglected sleep, and the joy of painting with Bob Ross later, I realised that I let this rework get a little out of hand. Feels pretty good ingame, though.

Removes delay from recoil, refactors large swaths of recoil code, removes multiple aiming cursor bugs (and potentially adds new ones), reworks armor slowdown into a combination of slowdown and "stiffness", a new stat that causes armor to give you recoil as you walk, a new stat that causes helmets to provides extra base offset (finally we can reduce our dependence on the publicly hated TINT value).

A couple minor armor value tweaks where I realised I left plotholes in previous PRs (riot mask), and a new fix to armor plates that prevents them from being placed on already armoured clothing (overcoats, armor vests, etc.), as well as gives more hefty debuffs to them.

Hopefully I didn't forget anything.

## Why It's Good For The Game

Less bug, more fleshed out recoil, less annoying slowdown and tint.

Thanks to a few cheaty lines of code, the recoil of guns has been adjusted to be in effect identical to before the change. This means we get all the benefits of a better feel for recoil, without the drawback of balance issues.

Whether the style PR goes in or not is not a concern for this PR's balance changes.

This PR makes the armor stacking PR ready to test, as it addresses most balance issues brought by it.
The armor stacking PR is not ready to be merged live by merging this alone, due to potential risks with One Star limbs, and other potentially overpowered items.

## Changelog
:cl:
balance: All armor and clothing now provide less slowdown, but run and gunning in heavy armor reduces accuracy
balance: Recoil now decreases immediately after firing, the recoil of every weapon adjusted accordingly
balance: Thanks to reducing the dependence on TINT, much less guns will obscure your vision
/:cl:

### CODE REVIEW

File: code/modules/mob/living/recoil.dm [hunk 1]
```diff
@@ -1,43 +1,89 @@
 /mob/living/proc/handle_recoil(var/obj/item/gun/G)
 	deltimer(recoil_reduction_timer)
-	if(G.one_hand_penalty)//If the gun has a two handed penalty and is not weilded.
+	if(G.one_hand_penalty) // If the gun has a two handed penalty and is not weilded.
```
Comment (mattroks101) [inline code/modules/mob/living/recoil.dm:3]: I know this is really superfluous but "weilded" is driving me crazy. Can you fix it to "wielded"?

File: tools/hooks/Install.bat [hunk 2]
```diff
@@ -1,2 +0,0 @@
-@call "%~dp0\..\bootstrap\python" -m hooks.install %*
```
Comment (mattroks101) [inline tools/hooks/Install.bat:1]: ???

Comment (Humonitarian) [reply tools/hooks/Install.bat:1]: What

Comment (mattroks101) [reply tools/hooks/Install.bat:1]: Is this supposed to be touched?

Comment (Humonitarian) [reply tools/hooks/Install.bat:1]: Not really, no

File: code/modules/mob/living/recoil.dm (hunk 1 above)
Comment (Humonitarian) [reply code/modules/mob/living/recoil.dm]: I did you extra too

File: code/game/objects/items/wielding.dm [hunk 3]
```diff
@@ -4,7 +4,7 @@
 	var/wielded = FALSE //Whether or not it's wielded.
 	var/wieldsound = 'sound/weapons/thudswoosh.ogg' //Generic sound. Replace it with a special one if you have one.
 	var/unwieldsound //If you want it to make a sound when you unwield, put one here.
-	var/wielded_icon //The item state used when it's weilded. Guns are snowflakey and have their own shit for this. This is for non guns.
+	var/wielded_icon //The item state used when it's wielded. Guns are snowflakey and have their own shit for this. This is for non guns.
```
Comment (mattroks101) [inline code/game/objects/items/wielding.dm:7]: Thanks I love you <3

**Pull request merged**